### ISSUE TITLE
Publish Docker images on DockerHub

### ISSUE THREAD

**Issue by SebastianBezold** [eclipse-tractusx/knowledge-agents#18]
## TL;DR

- Build workflow needs fixing, so the Docker build and publish process is successfully run
- Suggestion: choose more descriptive names for Docker images
- Add "Notice for Docker Images" for each image build by this repo. Description in [TRG 4.06](https://eclipse-tractusx.github.io/docs/release/trg-4/trg-4-06)
- Clarify legal info on `ontop/ontop` for ["provisioning" image](https://github.com/eclipse-tractusx/knowledge-agents/blob/main/provisioning/src/main/docker/Dockerfile)

## Description 

As described in [TRG 4.05](https://eclipse-tractusx.github.io/docs/release/trg-4/trg-4-05/), all Tractus-X Docker images must be published on DockerHub. The [current workflow](https://github.com/eclipse-tractusx/knowledge-agents/blob/main/.github/workflows/build.yml), that is supposed to build and publish the images, did not have a successful run yet. 

Besides the successful run, there is a critical step missing: 
[TRG 4.06](https://eclipse-tractusx.github.io/docs/release/trg-4/trg-4-06) is describing necessary legal documentation, that must be included in the repository and also used as description on the images DockerHub pages. Since the knowledge-agents repository does contain multiple Docker image, I would suggest to place dedicated Notices as .md file directly beside the Dockerfiles. This makes it easier to annotate different base images.

The [ontop/ontop](https://hub.docker.com/r/ontop/ontop) image is also not yet added to the aligned base image list of [TRG 4.02](https://eclipse-tractusx.github.io/docs/release/trg-4/trg-4-02). Looking at the [GitHub repo](https://github.com/ontop/ontop), the License should be fine, but we need to collect the necessary info for the legal notice

blocks #5

**Comment by SebastianBezold** [comment]
Legal notice for [tomcat:9-jre-11-temurin](https://github.com/eclipse-tractusx/knowledge-agents/blob/main/remoting/src/main/docker/Dockerfile#L48) used by "remoting" should also be aligned for legal notice

**Comment by SebastianBezold** [comment]
Hi @drcgjung, I think this issue here is the most pressing to fix before we are able to create a release. 
This would include:

-  fixing the build workflow,
- adding the legal information ("Notice for Docker image") for the used base images
- publish images with legal info to DockerHub

Only when these steps are done, we can start adding the helm test workflows and so on
